src/entities/join/index.ts

```diff
@@ -0,0 +1,86 @@
+import { Address } from 'viem';
+import { TokenAmount } from '../tokenAmount';
+import { Slippage } from '../slippage';
+
+export enum JoinKind {
+    Init = 'Init',
+    Proportional = 'Proportional',
+    Unbalanced = 'Unbalanced',
+    SingleAsset = 'SingleAsset',
+    ExactOut = 'ExactOut',
+}
+
+// Returned from API and used as input
+export type PoolState = {
+    id: Address;
+    address: Address;
+    type: string;
+    // TODO: is it ok to replace by Token here? Or should we stick to basic types?
+    tokens: {
+        address: Address;
+        decimals: number;
+    }[]; // already properly sorted in case different versions sort them differently
```

Comment:
Might want to include the `index` here and not rely on the tokens already being sorted as it creates a hidden requirement.

Reply:
I think here we were working on assumption that it will be leveraging the API to provide information in correct format, etc but I guess we're still working out limits.